File: src/main/java/io/scriptor/fdt/Tree.java
package io.scriptor.fdt;

import org.jetbrains.annotations.NotNull;

import java.nio.ByteBuffer;

import static io.scriptor.fdt.Constant.FDT_END;

public record Tree(@NotNull Node root) {

    public void write(final @NotNull ByteBuffer buffer, final @NotNull StringTable strings) {
        root.write(buffer, strings);
        buffer.putInt(FDT_END);
    }

    public int size() {
        int size = 0;
        size += root.size();
        size += Integer.BYTES;
        return size;
    }
}

File: src/main/java/io/scriptor/fdt/Tree.kt
package io.scriptor.fdt

import java.nio.ByteBuffer

data class Tree(val root: Node) {
    fun write(buffer: ByteBuffer, strings: StringTable) {
        root.write(buffer, strings)
        buffer.putInt(Constant.FDT_END.toInt())
    }

    fun size(): Int {
        var size = 0
        size += root.size()
        size += Integer.BYTES
        return size
    }
}
